src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -190,5 +190,13 @@ public ISet<LibraryRange> GetAllUnresolved()
                 }
             }
         }
+
+        protected async Task CommitCacheFileAsync(ILogger log)
+        {
+            log.LogDebug($"Writing cache file to disk. Path: {CacheFilePath}");
```

Comment:
If the restore failed the cache file should be deleted instead of writing out the current hash. Potentially a user could add the needed packages to their source and restore would continue nooping without picking them up.